include/circt/Dialect/Calyx/CalyxPrimitives.td

```diff
@@ -93,3 +93,52 @@ def MemoryOp : CalyxPrimitive<"memory", [
     )>
   ];
 }
+
+class CalyxLibraryOp<string mnemonic, list<OpTrait> traits = []> :
+  CalyxPrimitive<"lib_" # mnemonic, traits> {
+
+  let summary = "Defines an operation which maps to a Calyx library primitive";
+  let description = [{
+    This operation represents an instance of a Calyx library primitive.
+    A library primitive maps to some hardware-implemented component within the
+    native Calyx compiler.
+  }];
+
+  let arguments = (ins StrAttr:$instanceName);
+  let builders = [
+    OpBuilder<(ins "StringAttr":$instanceName, "::mlir::TypeRange":$resultTypes), [{
+      $_state.addAttribute("instanceName", instanceName);
+      $_state.addTypes(resultTypes);
+    }]>
+  ];
+}
+
+class BoolBinLibOp<string mnemonic> : CalyxLibraryOp<mnemonic, [
```

Comment:
Super nit (feel free to ignore): sooo much shorthand. I like verbosity :-)

Reply:
That does look a bit like shorthand overuse :eyes: - will fix!